src/metabase/api/tiles.clj

```diff
@@ -178,26 +177,38 @@
    lat-field   s/Str
    lon-field   s/Str
    query       su/JSONString}
-  (let [zoom        (Integer/parseInt zoom)
-        x           (Integer/parseInt x)
-        y           (Integer/parseInt y)
+  (let [zoom          (Integer/parseInt zoom)
+        x             (Integer/parseInt x)
+        y             (Integer/parseInt y)
+        lat-field-ref (field-ref lat-field)
+        lon-field-ref (field-ref lon-field)
 
         query
         (normalize/normalize (json/parse-string query keyword))
 
         updated-query (query->tiles-query query {:zoom zoom :x x :y y
-                                                 :lat-field lat-field
-                                                 :lon-field lon-field})
+                                                 :lat-field lat-field-ref
+                                                 :lon-field lon-field-ref})
 
-        {:keys [status], {:keys [rows]} :data, :as result}
+        {:keys [status], {:keys [rows cols]} :data, :as result}
         (qp/process-query-and-save-execution! updated-query
                                               {:executed-by api/*current-user-id*
-                                               :context     :map-tiles})]
-    ;; manual ring response here.  we simply create an inputstream from the byte[] of our image
+                                               :context     :map-tiles})
+
+        lat-key (qp.util/field-ref->key lat-field-ref)
+        lon-key (qp.util/field-ref->key lon-field-ref)
+        find-fn (fn [lat-or-lon-key]
+                  (first (keep-indexed
+                          (fn [idx col] (when (= (qp.util/field-ref->key (:field_ref col)) lat-or-lon-key) idx))
+                          cols)))
+        lat-idx (find-fn lat-key)
+        lon-idx (find-fn lon-key)
+        points  (for [row rows]
+                  [(nth row lat-idx) (nth row lon-idx)])]
     (if (= status :completed)
       {:status  200
        :headers {"Content-Type" "image/png"}
-       :body    (ByteArrayInputStream. (tile->byte-array (create-tile zoom rows)))}
+       :body    (tile->byte-array (create-tile zoom points))}
```

Comment:
I removed the conversion to `ByteArrayInputStream` temporarily while writing the new test, then realized that the frontend was still displaying the map as expected, so I figured it's not necessary. I don't really understand why it was there in the first place though so please correct me if anyone knows better.

Reply:
I think I didn't realize that Compojure/Ring would handle a byte array response but I knew it would handle an InputStream which is why I wrapped it. So if it works then 👍, let's remove it